opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -166,16 +166,40 @@
         SpanBuilder withStartTimestamp(long microseconds);
 
         /**
+         * @deprecated use {@link #startActive} instead.
          * Returns a newly started and activated {@link Scope}.
          *
          * <p>
-         * The returned {@link Scope} supports try-with-resources. For example:
+         * Note: {@link SpanBuilder#startActive(boolean)} is a shorthand for
+         * {@code tracer.scopeManager().activate(spanBuilder.start(), finishSpanOnClose)}.
+         *
+         * @param finishSpanOnClose whether span should automatically be finished when {@link Scope#close()} is called
+         * @return a {@link Scope}, already registered via the {@link ScopeManager}
+         *
+         * @see ScopeManager
+         * @see Scope
+         */
+        @Deprecated
+        Scope startActive(boolean finishSpanOnClose);
+
+        /**
+         * Returns a newly started and activated {@link Scope}.
+         *
+         * The span will not be automatically finished when {@link Scope#close()} is called.
+         *
+         * <p>
+         * The returned {@link Scope} supports try-with-resources, but is usually
+         * used like this:
          * <pre><code>
-         *     try (Scope scope = tracer.buildSpan("...").startActive(true)) {
+         *     Scope scope = null;
+         *     try {
+         *         scope = tracer.buildSpan("...").startActive();
```

Comment:
please inline this before the try. That removes the null checking later.

Also, as mentioned I think the scope having an accessor to the span is problematic.